Sources/web3swift/Web3/Web3+GasOracle.swift

```diff
@@ -0,0 +1,220 @@
+                .compactMap { t -> EthereumTransaction? in
+                    guard case let .transaction(transaction) = t else { return nil }
+                    return transaction
+                }
+                // TODO: Add filter for transaction types
+                .map { $0.maxPriorityFeePerGas }
+
+            return try calculateStatistic(transactionsTips, statistic)
+        }
+
+        private func suggestBaseFee(_ statistic: Statistic) throws -> BigUInt {
+            let latestBlockNumber = try eth.getBlockNumber()
+
+            // Assigning last block to object var to predict baseFee of the next block
+            latestBlock = try eth.getBlockByNumber(latestBlockNumber)
+            // TODO: Make me work with cache
+            let lastNthBlocksBaseFees = try (latestBlockNumber - blocksCount ... latestBlockNumber)
+                .map { try eth.getBlockByNumber($0) }
+                .filter { !$0.transactions.isEmpty }
+                .map { $0.baseFeePerGas }
+
+            return try calculateStatistic(lastNthBlocksBaseFees, statistic)
+        }
+
+        private func suggestGasFeeLegacy(_ statistic: Statistic) throws -> BigUInt {
+            let latestBlockNumber = try eth.getBlockNumber()
+
+            // Assigning last block to object var to predict baseFee of the next block
+            latestBlock = try eth.getBlockByNumber(latestBlockNumber)
+            // TODO: Make me work with cache
+            let lastNthBlockGasPrice = try (latestBlockNumber - blocksCount ... latestBlockNumber)
+                .map { try eth.getBlockByNumber($0, fullTransactions: true) }
+                .flatMap { b -> [EthereumTransaction] in
+                    b.transactions.compactMap { t -> EthereumTransaction? in
+                        guard case let .transaction(transaction) = t else { return nil }
+                        return transaction
+                    }
+                }
+                .map { $0.gasPrice }
+
+            return try calculateStatistic(lastNthBlockGasPrice, statistic)
+        }
+    }
+}
+
+public extension Web3.Oracle {
+    // MARK: - Base Fee
+    /// Base fee amount based on last Nth blocks
+    ///
+    /// Normalized means that most high and most low value were droped from calculation.
+    ///
+    /// Nth block may include empty ones.
+    ///
+    /// - Parameter statistic: Statistic to apply for base fee calculation
+    /// - Returns: Suggested base fee amount according to statistic, nil if failed to perdict
+    func predictBaseFee(_ statistic: Statistic) -> BigUInt? {
+        guard let value = try? suggestBaseFee(statistic) else { return nil }
+        return value
+    }
+
+    // MARK: - Tip
+    /// Maximum tip amount based on last block tips
+    ///
+    /// Normalized means that most high and most low value were droped from calculation.
+    ///
+    /// Method will takes in accounting a latest block that have transactions included more than `transactionsCount` property.
```

Comment:
this is still unclear as to what you are trying to say.

Are you trying to say this method calculates the suggested tip based on the most recent block that contains more than transactionsCount transactions?

Reply:
I've copied your term, thanks, this one is hard to tell

Reply:
English is hard, even for those of us that speak it natively! :)